libs/ledger-live-common/src/bot/index.ts

```diff
@@ -105,39 +99,72 @@ function makeAppJSON(accounts: Account[]) {
   return JSON.stringify(jsondata);
 }
 
-export async function bot({ currency, family, mutation }: Arg = {}): Promise<void> {
-  const SEED = getEnv("SEED");
-  invariant(SEED, "SEED required");
+export function getSpecs({ disabled, filter }) {
   const specs: any[] = [];
-  const specsLogs: string[][] = [];
-  const maybeCurrency = currency ? findCryptoCurrencyByKeyword(currency) : undefined;
-  const maybeFilterOnlyFamily = family;
+  const filteredCurrencies = filter?.currencies || [];
+  const filteredFamilies = filter?.families || [];
+  const filteredMutation = filter?.mutation;
+  let disabledCurrencies = disabled?.currencies || [];
+  let disabledFamilies = disabled?.families || [];
+
+  if (filteredFamilies.length > 0) {
+    // We want to ignore disabled families when user filters on a family
+    disabledFamilies = [];
+  }
+
+  if (filteredCurrencies.length > 0) {
+    // We want to ignore disabled currencies when user filters on a currency
+    disabledCurrencies = [];
+  }
 
   for (const family in allSpecs) {
-    const familySpecs = allSpecs[family];
-    if (maybeFilterOnlyFamily && maybeFilterOnlyFamily !== family) {
+    if (filteredFamilies.length > 0 && !filteredFamilies.includes(family)) {
+      // We only want to test specific families when we use a filter
+      continue;
+    }
+
+    if (disabledFamilies.includes(family)) {
+      // We don't want to test disabled families
       continue;
     }
 
+    const familySpecs = allSpecs[family];
+
     for (const key in familySpecs) {
       let spec: AppSpec<any> = familySpecs[key];
+      if (filteredCurrencies.length > 0 && !filteredCurrencies.includes(spec.currency.id)) {
```

Comment:
I think you removed the "findCryptoCurrencyByKeyword" logic from previous code.
The strings in "filteredCurrencies" are not necessary currency id, they can be tick, id, name or keywords.
Maybe you can preprocess filteredCurrencies array with "findCryptoCurrencyByKeyword"

Reply:
Yes,  but I think we should filter by currency id and family names

We know what we want to test

To be discussed